pyquil/quil.py

```diff
@@ -186,6 +192,7 @@ def copy_everything_except_instructions(self) -> "Program":
             # TODO: remove this type: ignore once rpcq._base.Message gets type hints.
             new_prog.native_quil_metadata = self.native_quil_metadata.copy()  # type: ignore
         new_prog.num_shots = self.num_shots
+        new_prog._memory = self._memory.copy()
         return new_prog
 
     def copy(self) -> "Program":
```

Comment:
Do we want to copy a fresh set `Memory` instance here too?

Reply:
It's done in `copy_everything_exept_instructions`